staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -104,6 +104,19 @@ func NewEventFromRequest(req *http.Request, level auditinternal.Level, attribs a
 	return ev, nil
 }
 
+// UpdateEventAttribs updates authorizer attributes according to attribs
+func UpdateEventAttribs(ev *auditinternal.Event, attribs authorizer.Attributes) {
+	if user := attribs.GetUser(); user != nil {
+		ev.User.Username = user.GetName()
```

Comment:
Usually EffectiveUser is equal ImpersonatedUser.
But when user run kubectl with `--as=username` option, the difference comes.

This tables shows the difference, we can see that some groups are automatically added.
![a](https://user-images.githubusercontent.com/20855209/28196203-9cc69e44-6881-11e7-8bc3-35ab5cf5b0a9.png)

The related codes are here:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go#L87
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/filters/impersonation.go#L121